src/goalServices/changeGoalStatus.ts

```diff
@@ -48,4 +48,8 @@ export default async function changeGoalStatus({
     reason,
     context,
   });
+
+  await goal.reload();
```

Comment:
The goal needs to be reloaded here because the `afterCreate` hook that fires on the `GoalStatusChange` table updates the goal at this point, so we want to make sure we return the newly-updated goal and not the old, stale one.